src/bin/pg_autoctl/cli_show.c

```diff
@@ -152,6 +156,14 @@ typedef struct ShowFileOptions
 
 static ShowFileOptions showFileOptions;
 
+typedef struct ShowUriOptions
+{
+	bool monitorOnly;
+	char formation[NAMEDATALEN];
+} ShowUriOptions;
```

Comment:
What about also having a `PostgresSetup pgSetup` slot in there and then using it in the option parsing function, so that we can later initialise easily either a monitor or a keeper setup depending on what we find from the PGDATA, rather than having to transform the `keeperOptions` into a monitor setting?

If that works well, I'm thinking we should at some point prepare a refactoring for all the other “dual mode” commands to use this technique rather than the current `monitor_config_init_from_pgsetup` way.

Reply:
We would still use `monitor_config_init_from_pgsetup` I guess, but from a "clean" pgSetup rather than from the keeperOptions ; it would allow to decouple the option parsing data structures from the keeper and monitor internal state data structures, and I think it would be good to decouple those.

Reply:
Off line discussion shows that this kind of refactoring needs more design still so that we can do a full command option parsing facility rather than try to solve it partially just here.